src/devices/SM28BYJ48/SM28BYJ48.cs

```diff
@@ -0,0 +1,202 @@
+        /// </summary>
+        /// <param name="pin1">The GPIO pin number which corresponds pin A on ULN2003 driver board.</param>
+        /// <param name="pin2">The GPIO pin number which corresponds pin B on ULN2003 driver board.</param>
+        /// <param name="pin3">The GPIO pin number which corresponds pin C on ULN2003 driver board.</param>
+        /// <param name="pin4">The GPIO pin number which corresponds pin D on ULN2003 driver board.</param>
+        public SM28BYJ48(int pin1, int pin2, int pin3, int pin4)
+        {
+            _pin1 = pin1;
+            _pin2 = pin2;
+            _pin3 = pin3;
+            _pin4 = pin4;
+
+            _controller = new GpioController();
+
+            _controller.OpenPin(_pin1, PinMode.Output);
+            _controller.OpenPin(_pin2, PinMode.Output);
+            _controller.OpenPin(_pin3, PinMode.Output);
+            _controller.OpenPin(_pin4, PinMode.Output);
+        }
+
+        /// <summary>
+        /// Set the stepper's mode.
+        /// </summary>
+        /// <param name="type">The mode.</param>
+        /// <returns>Current stepper motor.</returns>
+        public SM28BYJ48 SetStepperType(StepperType type)
+        {
+            _type = type;
+            return this;
+        }
+
+        /// <summary>
+        /// Set the stepper's delay. 
+        /// </summary>
+        /// <param name="ms">The milliseconds value.</param>
+        /// <returns>Current stepper motor.</returns>
+        public SM28BYJ48 SetStepperDelay(int ms)
+        {
+            _delayMs = ms > 1 ? ms : 1;
+            return this;
+        }
+
+        /// <summary>
+        /// This method is being called when the motor has completed the amount of steps.
+        /// </summary>
+        /// <param name="movementCompleted">The callback.</param>
+        /// <returns>Current stepper motor.</returns>
+        public SM28BYJ48 AddMovementCompletionHandler(Action movementCompleted)
+        {
+            _movementCompleted = new Action(movementCompleted);
+            return this;
+        }
+
+        /// <summary>
+        /// Stop the motor.
+        /// </summary>
+        public void Stop()
+        {
+            _rotationDirection = RotationDirection.Stopped;
+            _controller.Write(_pin1, PinValue.Low);
+            _controller.Write(_pin2, PinValue.Low);
+            _controller.Write(_pin3, PinValue.Low);
+            _controller.Write(_pin4, PinValue.Low);
+        }
+
+        /// <summary>
+        /// Rotate the motor clockwise.
+        /// </summary>
+        /// <param name="steps">Number of steps.</param>
+        public void RotateClockwise(int steps)
+        {
+            SetRotationDirection(RotationDirection.Clockwise, steps);
+            Start();
+        }
+
+        /// <summary>
+        /// Rotate the motor counter clockwise.
+        /// </summary>
+        /// <param name="steps">Number of steps.</param>
+        public void RotateCounterClockwise(int steps)
```

Comment:
make the RotateSteps take negative numbers and that will reduce it to a single API

Reply:
Fixed